apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns.php

```diff
@@ -45,35 +69,26 @@ public static function get_instance() {
 	/**
 	 * Register FSE block patterns and categories.
 	 */
-	public function register_patterns() {
-		// Register Block Pattern Categories.
-		if ( class_exists( 'WP_Block_Pattern_Categories_Registry' ) ) {
-			register_block_pattern_category( 'blog', array( 'label' => _x( 'Blog', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'call-to-action', array( 'label' => _x( 'Call to Action', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'contact', array( 'label' => _x( 'Contact', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'images', array( 'label' => _x( 'Images', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'list', array( 'label' => _x( 'List', 'Block pattern category', 'full-site-editing' ) ) );
-
-			// The 'Two Columns of Text' pattern is in the 'columns' and 'text' categories.
-			// Removing 'columns' so it doesn't appear as a category with only a single item.
-			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'columns' ) ) {
-				unregister_block_pattern_category( 'columns' );
-			}
-		}
+	private function register_patterns() {
+		$block_patterns = $this->get_patterns();
 
-		if ( class_exists( 'WP_Block_Patterns_Registry' ) ) {
-			// Remove core patterns except 'Two Columns of Text'.
-			// Unfortunately, \WP_Block_Patterns_Registry::get_instance()->get_all_registered() doesn't return the pattern names as keys.
-			foreach ( \WP_Block_Patterns_Registry::get_instance()->get_all_registered() as $pattern ) {
-				if ( 'core/' === substr( $pattern['name'], 0, 5 ) && 'core/text-two-columns' !== $pattern['name'] ) {
-					unregister_block_pattern( $pattern['name'] );
-				}
+		foreach ( (array) $this->get_patterns() as $pattern ) {
+			foreach ( (array) $pattern['tags'] as $slug => $tag ) {
+				register_block_pattern_category( $slug, array( 'label' => $tag['title'] ) );
 			}
 
-			// Add our patterns.
-			foreach ( $this->get_patterns() as $name => $pattern ) {
-				register_block_pattern( $name, $pattern );
-			}
+			register_block_pattern(
+				Block_Patterns::PATTERN_NAMESPACE . $pattern['name'],
+				array(
+					'title'         => $pattern['title'],
+					'description'   => $pattern['title'],
```

Comment:
Is the doubling up of `$pattern['title']` here for the description intended? We do have a `description` field in the endpoint, that pulls from `post_excerpt` in the original patterns, but I don't believe any of the patterns are currently using it.

Reply:
I think this probably makes sense to use title twice.

Also not sure we translate the description yet. (it might be that we do but filter when empty)